wordpress-mobile/WordPress-iOS
# Issue/9433 Reader Site Search

Fixes #9433. This PR extends Reader search to allow searching of sites as well as posts. Results should roughly match the results in the Reader in Calypso (which has a Sites section on the right side of the search results).

![reader-site-search-1](https://user-images.githubusercontent.com/4780/41180960-f5d14f40-6b67-11e8-881e-38a216370c24.gif)

The majority of the changes are a new service, `ReaderSiteSearchService` and a new view controller, `ReaderSiteSearchServiceViewController`. There are also changes to `ReaderSearchViewController` to accommodate the new search.

**To test:**

* Build and run.
* Navigate to Reader > Search
* Try some searches!
* Check that switching between Posts and Sites works correctly.
* Check that infinite scrolling works correctly in the posts list. Try a search that returns a lot of results (like `WordPress`), and also one that only returns a few pages – I found that `splatoon team` should only return 3 pages, so you should see infinite scrolling stop after 3 pages.
* Check searching when you have no connectivity.
* Try a search that has no results.
* Try a search that only has one results – for example: `tiredoldhack`.
